### PR TITLE
Use PKCS5Padding instead of ISO10126Padding for secrets encryption

### LINKED ISSUE #14153: In 4.3.9 if using the Oracle JDK Certain Authentication types fail.
* Graylog Version: 4.3.9
* Java Version: 1.8.0_261 Oracle

After updating to 4.3.9, when attempting login with a previously setup Authentication Provider the below error is thrown.

2022-12-05T11:32:21.149+04:00 INFO [SessionCreator] Invalid credentials in session create request. Actor: "urn:graylog:user:USERNAME"
2022-12-05T11:32:26.872+04:00 ERROR [AESTools] Could not encrypt value.
java.lang.SecurityException: JCE cannot authenticate the provider BC
        at javax.crypto.Cipher.getInstance([Cipher.java:660](http://cipher.java:660/)) ~[?:1.8.0_271]

        at [org.graylog2.security](http://org.graylog2.security/).AESTools.encrypt([AESTools.java:63](http://aestools.java:63/)) ~[graylog.jar:?]

        at [org.graylog2.security](http://org.graylog2.security/).encryption.EncryptedValueService.encrypt([EncryptedValueService.java:45](http://encryptedvalueservice.java:45/)) ~[graylog.jar:?]

        at [org.graylog2.security](http://org.graylog2.security/).realm.UsernamePasswordRealm.doGetAuthenticationInfo([UsernamePasswordRealm.java:92](http://usernamepasswordrealm.java:92/)) ~[graylog.jar:?]

        at [org.graylog2.security](http://org.graylog2.security/).realm.UsernamePasswordRealm.doGetAuthenticationInfo([UsernamePasswordRealm.java:71](http://usernamepasswordrealm.java:71/)) ~[graylog.jar:?]

        at org.apache.shiro.realm.AuthenticatingRealm.getAuthenticationInfo([AuthenticatingRealm.java:571](http://authenticatingrealm.java:571/)) ~[graylog.jar:?]

        at org.apache.shiro.authc.pam.ModularRealmAuthenticator.doMultiRealmAuthentication([ModularRealmAuthenticator.java:225](http://modularrealmauthenticator.java:225/)) ~[graylog.jar:?]

        at org.apache.shiro.authc.pam.ModularRealmAuthenticator.doAuthenticate([ModularRealmAuthenticator.java:275](http://modularrealmauthenticator.java:275/)) ~[graylog.jar:?]

        at org.apache.shiro.authc.AbstractAuthenticator.authenticate([AbstractAuthenticator.java:198](http://abstractauthenticator.java:198/)) ~[graylog.jar:?]

        at org.apache.shiro.mgt.AuthenticatingSecurityManager.authenticate([AuthenticatingSecurityManager.java:106](http://authenticatingsecuritymanager.java:106/)) ~[graylog.jar:?]

        at org.apache.shiro.mgt.DefaultSecurityManager.login([DefaultSecurityManager.java:275](http://defaultsecuritymanager.java:275/)) ~[graylog.jar:?]

        at [org.apache.shiro.subject.support](http://org.apache.shiro.subject.support/).DelegatingSubject.login([DelegatingSubject.java:260](http://delegatingsubject.java:260/)) ~[graylog.jar:?]

        at [org.graylog2.shared.security](http://org.graylog2.shared.security/).SessionCreator.login([SessionCreator.java:87](http://sessioncreator.java:87/)) ~[graylog.jar:?]

        at [org.graylog2.rest](http://org.graylog2.rest/).resources.system.SessionsResource.newSession([SessionsResource.java:142](http://sessionsresource.java:142/)) ~[graylog.jar:?]

        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_261]

        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_261]

        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_261]

        at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_261]

        at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0([ResourceMethodInvocationHandlerFactory.java:52](http://resourcemethodinvocationhandlerfactory.java:52/)) ~[graylog.jar:?]

### PR DESCRIPTION
**Pull request by mpfz0r**
If Graylog is run in an FIPS environment there is no crypto provider available that supports `AES/CBC/ISO10126Padding`. This is likely because this padding standard was withdrawn from ISO in 2007.
It is considered a bad practice to leave a subliminal channel in the padding.

We tried to workaround this by explicitly using BouncyCastle for the encryption/decryption.

This however creates problems with Oracle Java, because we strip the signature off the bouncy castle jar while repackaging it into our Uber-Jar.
In contrast to OpenJDK, Oracle Java does not allow the use of unsigned Security Providers.

Solution:

Change our secret encryption to using `PKCS5Padding` instead. There is a FIPS compatible provider `SunPKCS11-NSS-FIPS` available which supports `AES/CBC/PKCS5Padding`.

For backwards compatibility, we decrypt the ISO10126Padded keys without stripping the padding, and do that manually.

Fixes #14153

Refs #13525

### CODE REVIEW

**Review by bernd — APPROVED** [review]
Great work, thank you! :pray: 

I tested this with a 4.3 setup and LDAP. Without this change, I created the LDAP settings (and thus an encrypted value). Then I updated to a 4.3 that has this change applied. Everything worked.

**Comment by bernd** [comment]
I also tested on a FIPS-enabled RedHat.

File: changelog/unreleased/issue-14153.toml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+type = "fixed"
+message = "Use PKCS5 instead of ISO10126 padding for secrets encryption"
```
Comment (bernd) [inline changelog/unreleased/issue-14153.toml:2]: This is quite cryptic for regular users. How about something like "Improve JVM security provider compatibility by switching to a widely supported cipher transformation."

**Review by bernd — CHANGES_REQUESTED** [review]

**Review by thll — APPROVED** [review]
Looks good. Pending the one comment from @bernd, I think this is good to go.

Tested with RHEL 9 with fips mode disabled and also enabled. I was able to reproduce the issue with Oracle JDK 17 and Graylog version 5.0 and could observe the error being fixed when using the snapshot build for this PR. Reading encrypted values which have been written with Graylog version 5.0 worked flawlessly after the fix was applied.

**Review by bernd — APPROVED** [review]

**Comment by mpfz0r** [comment]
5.0.1 backport: https://github.com/Graylog2/graylog2-server/pull/14211
4.3.9 backport: https://github.com/Graylog2/graylog2-server/pull/14212

**Comment by boosty** [comment]
Thanks @mpfz0r for fixing, and @bernd and @thll for testing. Great work!

**Pull request merged**